addons/website/static/src/js/widgets/link_popover_widget.js

```diff
@@ -16,9 +16,18 @@ weWidgets.LinkPopoverWidget.include({
                 timeoutID = setTimeout(() => this.$target.popover('show'), 1500);
             });
         }
-
         return this._super(...arguments);
     },
+    async _loadAsyncLinkPreview() {
+        if(this.target.href){
+            let url = new URL(this.target.href);
+            if(url.protocol.startsWith('http') && window.location.hostname === url.hostname)
+                this.url = new URL(`/@${url.pathname}`, window.location.origin);
```

Comment:
What if the URL already has a `/@/` inside it?
![image](https://github.com/odoo/odoo/assets/30048408/bf2601cf-a979-4c5f-b539-f69061931e7e)